app/controllers/application_controller.rb

```diff
@@ -44,6 +48,18 @@ def infer_params_key(resource)
     klass.name.underscore.to_sym
   end
 
+# Sorting
+
+  # Sort by :id by default, so there is a consistent order even when no order has been
+  # selected via the interface
+  def default_sort_options
+    {id: :desc}
+  end
+
+  def sort(collection)
+    ModelSorter.sort(collection, params, default_sort_options)
```

Comment:
Suggestion:

```ruby
model_class = "#{collection.klass}Sorter".constantize rescue ModelSorter
model_class.sort(collection, params, default_sort_options)
```

So whenever you need to customise a sorter, like sorting by association, all you need to do is something like:

```ruby
class OrderSorter < ModelSorter

private
  def self.sort_scope(scope, attribute, direction)
    case attribute
    when "customer"
      scope.order("customers.name #{direction}")
    else
      super(scope, attribute, direction)
    end
  end
end
```

[PP] I'd just move ModelSorter to /app, but that's my personal preference...